packages/snowpack/src/commands/dev.ts

```diff
@@ -529,48 +536,69 @@ export async function command(commandOptions: CommandOptions) {
      * the URL. For example, "App.css" should return CSS but "App.css.proxy.js" should
      * return a JS representation of that CSS. This is handled in the wrap step.
      */
-    async function wrapResponse(code: string, hasCssResource: boolean) {
+    async function wrapResponse(
+      code: string,
+      {
+        hasCssResource,
+        sourceMap,
+        sourceMappingURL,
+      }: {hasCssResource: boolean; sourceMap?: string; sourceMappingURL: string},
+    ) {
+      // transform special requests
       if (isRoute) {
         code = wrapHtmlResponse({code: code, isDev: true, hmr: isHmr, config});
       } else if (isProxyModule) {
         responseFileExt = '.js';
-        code = await wrapImportProxy({
-          url: reqPath,
-          code,
-          isDev: true,
-          hmr: isHmr,
-          config,
-        });
-      } else if (responseFileExt === '.js') {
-        code = wrapImportMeta({code, env: true, isDev: true, hmr: isHmr, config});
-      }
-      if (responseFileExt === '.js' && hasCssResource) {
-        code = `import './${path.basename(reqPath).replace(/.js$/, '.css.proxy.js')}';\n` + code;
-      }
-      if (responseFileExt === '.js' && reqUrlHmrParam) {
-        code = await transformEsmImports(code as string, (imp) => {
-          const importUrl = path.posix.resolve(path.posix.dirname(reqPath), imp);
-          const node = hmrEngine.getEntry(importUrl);
-          if (node && node.needsReplacement) {
-            hmrEngine.markEntryForReplacement(node, false);
-            return `${imp}?${reqUrlHmrParam}`;
-          }
-          return imp;
-        });
+        code = await wrapImportProxy({url: reqPath, code, isDev: true, hmr: isHmr, config});
+      } else if (isSourceMap && sourceMap) {
+        code = sourceMap;
       }
-      if (responseFileExt === '.js') {
-        const isHmrEnabled = code.includes('import.meta.hot');
-        const rawImports = await scanCodeImportsExports(code);
-        const resolvedImports = rawImports.map((imp) => {
-          let spec = code.substring(imp.s, imp.e);
-          if (imp.d > -1) {
-            const importSpecifierMatch = spec.match(/^\s*['"](.*)['"]\s*$/m);
-            spec = importSpecifierMatch![1];
-          }
-          return path.posix.resolve(path.posix.dirname(reqPath), spec);
-        });
-        hmrEngine.setEntry(originalReqPath, resolvedImports, isHmrEnabled);
+
+      // transform other files
```

Comment:
A little bit of unplanned refactoring, but I had such a tough time finding this code block because it looked different from `build.ts`. Used a switch statement to organize the file transformations a bit better, and tried to start separating “meta” files (`.proxy.js`, `.map) from other non-meta files.